packages/testlab/src/test-sandbox.ts

```diff
@@ -0,0 +1,59 @@
+// Copyright IBM Corp. 2017,2018. All Rights Reserved.
+// Node module: @loopback/testlab
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {tmpdir} from 'os';
+import {createHash} from 'crypto';
+import {resolve, join, parse} from 'path';
+import * as util from 'util';
+import {mkdtempSync, mkdir as mkdir_, copyFile as copyFile_} from 'fs';
+const promisify = util.promisify || require('util.promisify/implementation');
+
+const rimraf = promisify(require('rimraf'));
+const mkdir = promisify(mkdir_);
+const copyFile = promisify(copyFile_);
+
+export class TestSandbox {
+  private path: string;
+
+  constructor() {
+    this.create();
+  }
+
+  private create(): void {
+    this.path = mkdtempSync(join(tmpdir(), 'lb4-temp-'));
+  }
+
+  private validateInst() {
+    if (!this.path) {
```

Comment:
TBH, I am not sure if this is really needed - when the directory was deleted, then underlying `fs` calls will report errors anyway. Are you concerned that these fs errors won't be helpful enough?

Not a big deal, your current implementation is fine with me.

Reply:
Yea the reason for a custom check was to ensure users don't get cryptic errors. If you believe this is not needed I am ok to remove it.